controllers/notes.js

```diff
@@ -0,0 +1,99 @@
+const note = require('../models/note');
+const { StatusCodes } = require('http-status-codes');
+const parseVErr = require('../utils/parseValidationErr');
+
+const getAllNotes = async (req, res) => {
+  const notes = await note.find({ createdBy: req.user.id }).sort('CreatedAt');
+  res.status(StatusCodes.OK).render('notes', { notes });
+};
+
+const addNote = async (req, res) => {
+  req.body.createdBy = req.user.id;
+  await note.create(req.body);
+  res.status(StatusCodes.CREATED);
+
+  const notes = await note.find({ createdBy: req.user.id }).sort('CreatedAt');
+  const info = [`The note has been successfully added.`];
+  res.render('notes', { notes, info });
+};
+
+const addForm = async (req, res) => {
+  res.render('note', { note: null });
+};
+
+const editForm = async (req, res) => {
+  const note = await note.findOne({
+    _id: req.params.id,
+    createdBy: req.user.id,
+  });
+
+  if (!note) {
```

Comment:
If the note is not found, you must display an error using the req.flash method and redirect to the "/notes" route.
```
if (!note) {
    req.flash("error", `No note with id ${req.params.id}`);
    return res.redirect('/notes');
  }
```